### PR TITLE
eksctl 0.24.0 (new formula)

### PR DESCRIPTION
**Pull request by crunchtime-ali**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?
-----

Here is yet another popular tool for DevOps aficionados.

### CODE REVIEW

File: Formula/eksctl.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  license "Apache-2.0"
+
+  depends_on "go" => :build
+  depends_on "go-bindata" => :build
+  depends_on "aws-iam-authenticator"
+  depends_on "kubernetes-cli"
+
+  def install
+    # Some dependencies expect go-bindata to be located in the GOPATH. We want to use the global one instead
+    inreplace "Makefile", "$(gopath)", HOMEBREW_PREFIX.to_s
```
Comment (dawidd6) [inline Formula/eksctl.rb:16]: `GOBIN` seems to be overridable in Makefile, maybe try something like this:

```suggestion
    ENV["GOBIN"] = HOMEBREW_PREFIX/"bin"
```

Comment (crunchtime-ali) [reply Formula/eksctl.rb]: @dawidd6 It works too

**Review by dawidd6 — DISMISSED** [review]

**Comment by chenrui333** [comment]
I know AWS docs is referencing the tap one, it might be good to communicate out to the tap repo and let them know that we can co-maintain this formula.

**Review by chenrui333 — DISMISSED** [review]
lgtm btw

File: Formula/eksctl.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  homepage "https://eksctl.io"
+  url "https://github.com/weaveworks/eksctl.git",
+      :tag      => "0.24.0",
+      :revision => "f23f8d7d75943d1a62c87ce2f3d455da7af741fa"
+  license "Apache-2.0"
+
+  depends_on "go" => :build
+  depends_on "go-bindata" => :build
+  depends_on "aws-iam-authenticator"
+  depends_on "kubernetes-cli"
```
Comment (chenrui333) [inline Formula/eksctl.rb:12]: This might be our value proposition.

But we probably dont have to specify the k8s dependency though.

**Comment by crunchtime-ali** [comment]
> I know AWS docs is referencing the tap one, it might be good to communicate out to the tap repo and let them know that we can co-maintain this formula.

For the `eksctl` documentation part I already created a branch to reference installing without the tap https://github.com/weaveworks/eksctl/compare/master...crunchtime-ali:switch-to-homebrew-core
I will create a PR when it lands, inform them that we will take care of it and mention your suggestion also to inform AWS.
Can you point me to the AWS documentation mentioning the installation?

File: Formula/eksctl.rb (hunk 2 above)
Comment (crunchtime-ali) [reply Formula/eksctl.rb]: I am with you on this one. I removed the dependency on `kubernetes-cli`

Comment (chenrui333) [reply Formula/eksctl.rb]: This is something I did before to migrate a tap formula to homebrew-core, `https://github.com/loadimpact/homebrew-k6/pull/7`, you might help it helpful.

**Review by chenrui333 — APPROVED** [review]

**Review by dawidd6 — APPROVED** [review]

**Pull request closed without merging**